### DISCUSSION TITLE
FuzzerDataProvider data for multiple variables

### DISCUSSION THREAD

**Discussion by manobaneru** [CodeIntelligenceTesting/jazzer#823] [Q&A] [answered]
Good evening! 

I decided to test the fuzzer on functions that take in several variables of different types. To see what the phaszer puts inside my variable each time, I decided to use the `System.out.println.` (That's probably a bad way of debugging, I realize that) And I noticed that every value of the boolean variable within 10000 iterations is always false. I additionally output the value of the variable `data `to the console and got `com.code_intelligence.jazzer.driver.FuzzedDataProviderImpl@477b4cdf` or `com.code_intelligence.jazzer.driver.FuzzedDataProviderImpl@477b4^C0`  on each iteration. What am I doing wrong?

Me code looks like this:
```
public class ClassName {
    public static void fuzzerTestOneInput(FuzzedDataProvider data) {
        byte[] s1 = data.consumeRemainingAsBytes();
        int i1 = data.consumeInt();
        int i2 = data.consumeInt();
        boolean b1 = data.consumeBoolean();
        System.out.println(data);
        System.out.println(i1);
        System.out.println(b1);
        System.out.println(i2);
        try {
            func1(s1, i1);
            func2(b1, s1, i1, i2, i1 ,i1, i1);
        } catch (ArrayIndexOutOfBoundsException e) {
        }
    }
}
```
And as a result on each iteration:
![image](https://github.com/CodeIntelligenceTesting/jazzer/assets/72031386/a6780179-40f0-44a5-bbff-ce97022a6b8f)

**Comment by fmeum** [comment, marked as answer]
All the `consumeRemaining*` functions consume the entire remaining fuzzer input, meaning that all calls that come after them will return constant default values. Could you try reordering your calls so that this function is invoked last?

**Reply by manobaneru** [reply]
It has helped, thank you! Now it looks like this:
![image](https://github.com/CodeIntelligenceTesting/jazzer/assets/72031386/fb4a2669-9536-445e-bb84-3085a1be8601)
And is it ok for the byte array to continue displaying this string? Is it worth trying to do something about it?

**Reply by br-lewis** [reply]
If your code is similar to what you first posted, you're printing `data` which is a `FuzzedDataProvider` and not `s1` which would be the byte array. If that's what you meant to print then, yes, this is expected as that isn't printing the contents of the fuzzer data, it's using Java's default Object [`toString`](https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#toString--).

**Reply by manobaneru** [reply]
Everything works now. I'm really grateful for your help!
